### ISSUE TITLE
no-shadow false positive when used with typescript

### ISSUE THREAD

**Issue by stephenotalora** [github/eslint-plugin-github#126]
hello there 👋🏽  

We're actively working to adopt `github/recommended` and `github/typescript` rules. Typescript enums are widely used in our repository, however, the recommended GitHub plugin yields a `no-shadow` error for every enum occurrence. 

**repro:**
```js
export enum DataType {
  Title = 'title',
  Text = 'text',
  Date = 'date',
  // ...
}

export enum SystemId {
  Title = 'Title',
   // ...
}
```

**Expected Result**
No lint error since enum has been defined just once.

**error(s):**
`<enum name>` is already declared in the upper scope (eslint no-shadow)

**version:** 

|package | version|
|-------- | -------|
|eslint-plugin-github| ^4.1.3|
|@typescript-eslint/eslint-plugin|^4.28.3|
|Typescript|^4.3.4|
|ESLint|^7.29.0|
|node|v14.17.1|

**Comment by stephenotalora** [comment]
Upon further investigation, this appears to be a widespread problem within the Typescript community.
- Example: https://github.com/typescript-eslint/typescript-eslint/issues?q=label%3A%22fix%3A+user+error%22+no-shadow
- There are suggestions to disable the eslint rule in favor of the `@typscript-eslint` counterpart please see https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code.

However, the latter overwrittes the rule that is recommended to use by this repo. There might be a collision between the two, What would the recommended usage for a typescript app? 

1. do we disable the rule?
2. do we only use `github/typescript` instead of `github/recommended`?

Feedback is much appreciated 🙏🏽

**Comment by stephenotalora** [comment]
Other false positive examples:

```
error  StyledComponentProps not found in 'styled-components'              import/named
error  DebouncedFunc not found in 'lodash-es'
```

named imports also causing false positive errors when used when used with `github/recommended` and `github/typescript`

**Comment by koddsson** [comment]
> * There are suggestions to disable the eslint rule in favor of the `@typscript-eslint` counterpart

I think this is exactly what we should do.

**Comment by koddsson** [comment]
> named imports also causing false-positive errors when used with `github/recommended` and `github/typescript`

I don't know what is causing this, though. Maybe we can pair on it when you got the time? A small reproduction of this problem would be great.

**Comment by stephenotalora** [comment]
> > named imports also causing false-positive errors when used with `github/recommended` and `github/typescript`
> 
> I don't know what is causing this, though. Maybe we can pair on it when you got the time? A small reproduction of this problem would be great.

@koddsson, a good example of how to repro, would be something like getting a type from a `@types/*` definition.  

**Repro:** 

```js
import styled, {StyledComponentProps} from 'styled-components'
```

the named import `{StyledComponentProps}` is getting picked as an error with an `import/name` failure:

```
error  StyledComponentProps not found in 'styled-components'  import/named
```

Note that this only seems to happen when we use `github/recommended` with `github/typescript`. Perhaps we can apply a similar fix as `no-shadow`? Example:

```
'import/named': 'off',
'@typescript-eslint/import/named': ['error'],
```

**Comment by stephenotalora** [comment]
opened a PR here if it helps https://github.com/github/eslint-plugin-github/pull/129

**Comment by stephenotalora** [comment]
Dropping a link here for reference: https://github.com/benmosher/eslint-plugin-import/blob/master/README.md#typescript, this should fix issues with the previous comment in this thread 🧵  Huge thanks @koddsson for helping fix this issue this week.
